sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -110,23 +155,165 @@ protected string GetSessionFilePath()
         /// This will run once before any tests.
         /// </summary>
         [OneTimeSetUp]
-        public void StartLoggingEvents()
+        public void InitializeRecordedTestClass()
         {
+            TestContext.Progress.WriteLine($"starting up {GetType().FullName}");
+
             if (Mode == RecordedTestMode.Live || Debugger.IsAttached)
             {
                 Logger = new TestLogger();
             }
+
+            if (!_useLegacyTransport)
+            {
+                StartTestProxy();
+            }
+        }
+
+        private void StartTestProxy()
+        {
+            var testProxyPath = GetType().Assembly.GetCustomAttributes<AssemblyMetadataAttribute>().Single(a => a.Key == "TestProxyPath")
+                .Value;
+            TestContext.Progress.WriteLine(testProxyPath);
+            ProcessStartInfo testProxyProcessInfo = new ProcessStartInfo(
+                s_dotNetExe,
+                testProxyPath)
+            {
+                UseShellExecute = false,
+                EnvironmentVariables =
+                    { { "ASPNETCORE_URLS", "http://0.0.0.0:0;https://0.0.0.0:0" }, { "Logging__LogLevel__Microsoft", "Information" } },
```

Comment:
127.0.0.1?

Reply:
I was seeing strange behavior when using this where the perf was much worse.